File: python/ex13.py
class Calculadora:
    @staticmethod
    def fatorial(valor):
        if valor < 0:
            raise ValueError("O fatorial não está definido para números negativos.")
        elif valor == 0 or valor == 1:
            return 1
        else:
            resultado = 1
            for i in range(2, valor + 1):
                resultado *= i
            return resultado

def executar():
    numero = 5
    resultado_fatorial = Calculadora.fatorial(numero)
    print(f"O fatorial de {numero} é: {resultado_fatorial}")

    try:
        fatorial_negativo = Calculadora.fatorial(-3)
    except ValueError as erro:
        print(erro)

if __name__ == "__main__":
    executar()

File: golang/ex13.go
package main

import (
	"fmt"
)

type Calculadora struct{}

func (Calculadora) Fatorial(n int) int {
	if n <= 1 {
		return 1
	}
	return n * Calculadora{}.Fatorial(n-1)
}

func main() {
	calculadora := Calculadora{}
	numero := 5
	resultado := calculadora.Fatorial(numero)
	fmt.Printf("Fatorial de %d é %d\n", numero, resultado)
}
